src/persist-client/src/lib.rs

```diff
@@ -225,12 +225,13 @@ impl PersistClient {
         trace!("Client::open shard_id={:?}", shard_id);
         let mut machine = Machine::new(shard_id, Arc::clone(&self.consensus)).await?;
         let (writer_id, reader_id) = (WriterId::new(), ReaderId::new());
-        let (write_cap, read_cap) = machine.register(&writer_id, &reader_id).await;
+        let (_, read_cap) = machine.register(&writer_id, &reader_id).await;
```

Comment:
IMO we should still return the upper from register instead of doing a fetch_upper on the next line

Reply:
> IMO we should still return the upper from register instead of doing a fetch_upper on the next line
:+1: